Modules/BarcodeScanner/BarcodeScannerViewModel.swift

```diff
@@ -0,0 +1,42 @@
+import JamitFoundation
+import UIKit
+
+/// The state view model for `BarcodeScannerView`.
+public struct BarcodeScannerViewModel: ViewModelProtocol {
+    /// The callback type for a successful barcode scan result.
+    public typealias ScanCallback = ([Barcode]) -> Void
+    /// The callback type for an error in the barcode scan process.
+    public typealias ErrorCallback = (BarcodeScannerError) -> Void
+
+    /// The barcode types which should be captured by the `BarcodeScannerView`.
+    public let barcodeTypes: [BarcodeType]
+    /// The callback for a successful barcode scan result.
+    public let onScan: ScanCallback
```

Comment:
Wouldn't `onSuccess` or `onScanResult` be a better name for this closure?

Reply:
Furthermore I would prefer to have them optional so `Scancallback?`and `ErrorCallback?`.

Reply:
You can already pass the properties optionally if you want to do we need to wrap them in an Optional?